api/api/utils/image_proxy/__init__.py

```diff
@@ -169,37 +169,35 @@ async def get(
         do_not_wait_for(
             _tally_response(tallies, media_info, month, domain, upstream_response)
         )
-        upstream_response.raise_for_status()
         status_code = upstream_response.status
         content_type = upstream_response.headers.get("Content-Type")
         logger.debug(
             "Image proxy response status: %s, content-type: %s",
             status_code,
             content_type,
         )
-        content = await upstream_response.content.read()
+        upstream_response.raise_for_status()
+        content = await upstream_response.read()
```

Comment:
This is interesting. There is a slight difference in `upstream_response.content.read()` and `upstream_response.read()`. The former (the original version) streams the content, whereas the new version (just calling `.read` on the response), reads the content entirely into memory.

I am pretty sure in this particular case there is no difference, though, because we weren't actually streaming the response back to the requester. We were streaming the whole response into memory and then writing it to the response later on. I think this is left over from when I tried to use `StreamingHttpResponse` and couldn't get it to work reliably. This should be an okay change to make! But I wanted to note it here _just in case_ there ends up being an issue during deployment that could be explained by this very minor and easy-to-miss difference in the code. I really think this is a safe change, though, and is positive because it uses the API that matches directly how we are actually using the response.